In [3]:
import pandas as pd
from pathlib import Path


In [ ]:
csv_path = Path.home() / "Downloads" /"How Much Sleep Do Americans"/"Time Americans Spend Sleeping.csv"
df_sleep = pd.read_csv(csv_path   )
df_sleep.head(50)

index  Year  Period  Avg hrs per day sleeping  Standard Error  \
0       0  2003  Annual                      8.57           0.018   
1       1  2004  Annual                      8.55           0.026   
2       2  2005  Annual                      8.62           0.023   
3       3  2006  Annual                      8.63           0.024   
4       4  2007  Annual                      8.57           0.024   
5       5  2008  Annual                      8.60           0.023   
6       6  2009  Annual                      8.67           0.023   
7       7  2010  Annual                      8.67           0.024   
8       8  2011  Annual                      8.71           0.026   
9       9  2012  Annual                      8.73           0.026   
10     10  2013  Annual                      8.74           0.027   
11     11  2014  Annual                      8.80           0.025   
12     12  2015  Annual                      8.83           0.027   
13     13  2016  Annual                      8.79           0.028   
14     14  2017  Annual                      8.80           0.027   
15     15  2003  Annual                      8.26           0.023   
16     16  2004  Annual                      8.25           0.034   
17     17  2005  Annual                      8.35           0.030   
18     18  2006  Annual                      8.33           0.032   
19     19  2007  Annual                      8.29           0.032   
20     20  2008  Annual                      8.30           0.032   
21     21  2009  Annual                      8.39           0.031   
22     22  2010  Annual                      8.38           0.031   
23     23  2011  Annual                      8.45           0.034   
24     24  2012  Annual                      8.45           0.034   
25     25  2013  Annual                      8.48           0.036   
26     26  2014  Annual                      8.54           0.031   
27     27  2015  Annual                      8.59           0.035   
28     28  2016  Annual                      8.50           0.037   
29     29  2017  Annual                      8.53           0.035   
30     30  2003  Annual                      9.29           0.026   
31     31  2004  Annual                      9.25           0.032   
32     32  2005  Annual                      9.28           0.037   
33     33  2006  Annual                      9.32           0.036   
34     34  2007  Annual                      9.24           0.035   
35     35  2008  Annual                      9.31           0.036   
36     36  2009  Annual                      9.33           0.031   
37     37  2010  Annual                      9.34           0.032   
38     38  2011  Annual                      9.35           0.040   
39     39  2012  Annual                      9.39           0.036   
40     40  2013  Annual                      9.34           0.039   
41     41  2014  Annual                      9.40           0.035   
42     42  2015  Annual                      9.40           0.041   
43     43  2016  Annual                      9.48           0.040   
44     44  2017  Annual                      9.44           0.035   
45     45  2003  Annual                      9.15           0.060   
46     46  2004  Annual                      9.23           0.081   
47     47  2005  Annual                      9.27           0.077   
48     48  2006  Annual                      9.19           0.064   
49     49  2007  Annual                      9.11           0.085   

                 Type of Days          Age Group  Activity   Sex  
0                    All days  15 years and over  Sleeping  Both  
1                    All days  15 years and over  Sleeping  Both  
2                    All days  15 years and over  Sleeping  Both  
3                    All days  15 years and over  Sleeping  Both  
4                    All days  15 years and over  Sleeping  Both  
5                    All days  15 years and over  Sleeping  Both  
6              

In [5]:
rows,cols = df_sleep.shape
print("Rows:", rows, "Columns:", cols)

Rows: 945 Columns: 9


In [6]:
df_sleep.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 945 entries, 0 to 944
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   index                     945 non-null    int64  
 1   Year                      945 non-null    int64  
 2   Period                    945 non-null    object 
 3   Avg hrs per day sleeping  945 non-null    float64
 4   Standard Error            945 non-null    float64
 5   Type of Days              945 non-null    object 
 6   Age Group                 945 non-null    object 
 7   Activity                  945 non-null    object 
 8   Sex                       945 non-null    object 
dtypes: float64(2), int64(2), object(5)
memory usage: 66.6+ KB


Dataset: How Much Sleep Do Americans Really Get?
1. This dataset contains 945 rows and 9 columns.
2. Columns include age group,types of days,average hours of sleep,etc.
3. This dataset helps to analyze sleeping habbits among Americans.

In [7]:
df_sleep.columns

Index(['index', 'Year', 'Period', 'Avg hrs per day sleeping', 'Standard Error',
       'Type of Days', 'Age Group', 'Activity', 'Sex'],
      dtype='object')

Columns: 
'index'                     - qualitative   - It’s just an identifier
'Year'                      - quantitative  - That's a number that represents a measurable value
'Period'                    - qualitative   - Data represented as a string (Annual)
'Avg hrs per day sleeping'  - quantitative  - Represents hours of sleep,measurable value
'Standard Error'            - quantitative  - Statistical value used to measure uncertainty(numeric)
'Type of Days'              - qualitative   - Categorical data ("Weekday", "Weekend")
'Age Group'                 - qualitative   - Categories "15-24", "15 and over"
'Activity'                  - qualitative   - Data describes activities (text labels)
'Sex'                       - qualitative   - Data is represented by a category (Male,Female,Both)


Interesting columns for analysis:

1. Year -> to see if Americans are sleeping more or less over time.

2. Avg hrs per day sleeping -> The main dependent variable; useful for distribution plots, mean/median comparisons, or correlation with other factors.

3. Sex -> Allows comparison of sleep patterns between men and women.

4.Age -> Group → To analyze whether younger vs. older adults differ in sleep duration.

 

